includes/class-wc-gateway-amazon-payments-advanced.php

```diff
@@ -619,11 +619,23 @@ public function maybe_separator_and_checkout_button() {
 	/**
 	 * Prints the Amazon Pay button on the product page when the setting is on.
 	 *
+	 * Additionally checks that the product is purchasable and not out of stock.
+	 *
 	 * @hooked on 'woocommerce_single_product_summary' on 35 priority
 	 *
 	 * @return void
 	 */
 	public function maybe_separator_and_checkout_button_single_product() {
+		$prd = wc_get_product();
```

Comment:
Not a stopper, but my suggestion to make the code more readable we need to use better names for variables. Eg. $product = wc_get_product();

Reply:
Done!